codegen/grid.py

```diff
@@ -717,6 +720,13 @@ def set_variable(self, var, value=0, type='int', constant=False):
             var = var.name
         self.defined_variable[var] = Variable(var, value, type, constant)
 
+    def set_page_size(self, page_size):
```

Comment:
I meant that I'm not sure how to allocate on heap using __attribute__, apart from using global variables (which I never used), the link seems to suggest using posix. I will add the generated code to the repo (src/tests/test3d.cpp), it looks something like this
```
float *_Txx_vec;
#ifdef _MSC_VER
_Txx_vec = (float*) _aligned_malloc(2315250*sizeof(float), 4096);
#else
posix_memalign((void **)(&_Txx_vec), 4096, 2315250*sizeof(float));
#endif
float (*Txx)[dim1][dim2][dim3]= (float (*)[dim1][dim2][dim3]) _Txx_vec;
```